srcbpatch/actionscollection.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+// #include "actionscollection.h"
+#include <string_view>
+#include <list>
+#include <memory>
+#include <vector>
+
+#include "dictionary.h"
+#include "jsonparser.h"
+
+namespace bpatch
+{
+/// <summary>
+///    class contains main entry points for processing.
+///    * JSON parser callback to load settings for processing
+///    * pipeline to process binary lexemes
+/// </summary>
+class ActionsCollection final: public TJsonCallBack, public StreamReplacer
+{
+public:
+    /// <summary>
+    ///   constructor accepts vector with json data and process it
+    /// </summary>
+    /// <param name="dataSource">json data - vecor data will be held inside and modified
+    ///   for inner simplicity of usage</param>
+    /// <param name="pLast">StreamReplacer to pass result of actions 
+    ///   in the end of processing</param>
+    ActionsCollection(std::vector<char>&& dataSource);
+
+    /// <summary>
+    ///  callback from TJsonCallBack
+    /// </summary>
+    /// <param name="pJson">Json object just got its name in parser</param>
+    void OnJsonObjectBegins(TJSONObject* const pJson) override;
+
+    /// <summary>
+    ///  callback from TJsonCallBack
+    /// </summary>
+    /// <param name="pJson">Json object just ends in parser</param>
+    void OnJsonObjectParsed(TJSONObject* const pJson) override;
+
+    /// <summary>
+    ///  callback from TJsonCallBack
+    /// </summary>
+    /// <param name="pJson">Json array just begins in parser</param>
+    void OnJsonArrayBegins(TJSONObject* const pJson) override;
+
+    /// <summary>
+    ///  callback from TJsonCallBack
+    /// </summary>
+    /// <param name="pJson">Json array just ends in parser</param>
+    void OnJsonArrayParsed(TJSONObject* const pJson) override;
+
+    /// <summary>
+    ///   callback from StreamReplacer
+    /// </summary>
+    /// <param name="toProcess">this char is under processing right now</param>
+    /// <param name="aEod">this is sign that no more data</param>
+    void DoReplacements(const char toProcess, const bool aEod) const override;
+
+    /// <summary>
+    ///   callback from StreamReplacer
+    /// </summary>
+    /// <param name="toProcess">replacer to call next</param>
+    void SetNextReplacer(StreamReplacer* const pNext) override;
+
+
+protected:
+    /// <summary>
+    ///   throws error if we meet error in the expected logic
+    /// </summary>
+    /// <param name="message">text of the error</param>
+    void ReportError(const char* const message);
+
+    /// <summary>
+    ///   throws error if we meet duplicate name of lexeme
+    /// </summary>
+    /// <param name="aname">duplicate name of lexeme</param>
+    void ReportDuplicateNameError(const std::string_view& aname);
+
+    /// <summary>
+    ///   throws error if we cannot find lexeme for composite lexeme
+    /// </summary>
+    /// <param name="aname">nonexistent name of lexeme</param>
+    void ReportMissedNameError(const std::string_view& aname);
```

Comment:
The three methods above doesn't use class. May be mark it as static or better to exclude from the class